drivers/net/Kconfig

```diff
@@ -72,6 +78,16 @@ config PPP_CLIENT_CLIENTSERVER
 	  This is only necessary if a ppp connection should be
 	  established with a Microsoft Windows PC.
 
+config PPP_DIALUP
+	bool "Features that are needed if using PPP for a dial up"
+	default n
+	depends on NET_SOCKETS_PACKET
+	depends on !NET_SOCKETS_PACKET_DGRAM
+	depends on !NET_L2_ETHERNET
```

Comment:
Why would we need to disable Ethernet if `PPP_DIAPUP` is enabled? Notice that there could be multiple network interfaces enabled in the system at the same time.

Reply:
It was due to: https://github.com/zephyrproject-rtos/zephyr/pull/33764/commits/c5d0815f76df65813b44c5d5662766492549f338#diff-4c42c7658751f4e7e7eac0be650e180c5571265da95850c00ce92f02858242deR64

I.e. ppp dialup need for the packet sckt is conflicting with the current packet sckt (used with ethernet flags as I understood), PPP needs L2 to be handled, current no. So, the intention here and in net_core flagging was that you can have either one but not the both at the same binary.  Unfortunately this is not providing full solution for that TODO #2, just to get this case working: https://github.com/nrfconnect/sdk-zephyr/blob/master/subsys/net/ip/packet_socket.c#L41

Reply:
Fixed: PPP_DIALUP config removed and made packet socket more generic and not using any PPP flags in ip/net core side.